subprojects/core/src/main/java/org/gradle/util/HasherUtil.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.util;
+
+import com.google.common.base.Charsets;
+import com.google.common.hash.Hasher;
+import org.apache.commons.lang.SerializationUtils;
+
+import java.io.NotSerializableException;
+import java.io.Serializable;
+import java.lang.reflect.Array;
+import java.math.BigInteger;
+import java.util.Map;
+
+public abstract class HasherUtil {
```

Comment:
No need to make the class `abstract`. Making it `final` plus a constructor with `private` modifier would be better.